File: specification/datamigration/resource-manager/Microsoft.DataMigration/DataMigration/stable/2025-06-30/examples-java/SqlDbGetDatabaseMigrationExpanded.java

/**
 * Samples for DatabaseMigrationsSqlDb Get.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/datamigration/resource-manager/Microsoft.DataMigration/DataMigration/stable/2025-06-30/examples/
     * SqlDbGetDatabaseMigrationExpanded.json
     */
    /**
     * Sample code: Get Sql DB database Migration with the expand parameter.
     * 
     * @param manager Entry point to DataMigrationManager.
     */
    public static void getSqlDBDatabaseMigrationWithTheExpandParameter(
        com.azure.resourcemanager.datamigration.DataMigrationManager manager) {
        manager.databaseMigrationsSqlDbs().getWithResponse("testrg", "sqldbinstance", "db1", null,
            "MigrationStatusDetails", com.azure.core.util.Context.NONE);
    }
}

File: specification/datamigration/resource-manager/Microsoft.DataMigration/DataMigration/preview/2025-09-01-preview/examples-js/SqlDbGetDatabaseMigrationExpanded.js
const { DataMigrationManagementClient } = require("@azure/arm-datamigration");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to retrieve the Database Migration resource.
 *
 * @summary retrieve the Database Migration resource.
 * x-ms-original-file: 2025-09-01-preview/SqlDbGetDatabaseMigrationExpanded.json
 */
async function getSqlDBDatabaseMigrationWithTheExpandParameter() {
  const credential = new DefaultAzureCredential();
  const subscriptionId = "00000000-1111-2222-3333-444444444444";
  const client = new DataMigrationManagementClient(credential, subscriptionId);
  const result = await client.databaseMigrationsSqlDb.get("testrg", "sqldbinstance", "db1", {
    expand: "MigrationStatusDetails",
  });
  console.log(result);
}
